OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseDatesPageFragment.kt

```diff
@@ -65,11 +84,46 @@ class CourseDatesPageFragment : OfflineSupportBaseFragment() {
         return binding.root
     }
 
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        when (requestCode) {
+            REQUEST_SHOW_COURSE_UNIT_DETAIL -> {
+                when (resultCode) {
+                    Activity.RESULT_OK -> {
+                        val outlineComp: CourseComponent? = courseManager.getCourseDataFromAppLevelCache(courseData.courseId)
+                        val leafCompId: String = data?.getSerializableExtra(Router.EXTRA_COURSE_COMPONENT_ID) as String
+                        val leafComp: CourseComponent? = courseManager.getComponentByIdFromAppLevelCache(courseData.courseId, leafCompId)
+                        val outlinePath = outlineComp?.path
+                        val leafPath = leafComp?.path
+                        val outlinePathSize = outlinePath?.path?.size ?: 0
+                        if (outlineComp != leafPath?.get(outlinePathSize - 1)) {
+                            activity?.setResult(Activity.RESULT_OK, data)
+                            activity?.finish()
+                        } else {
+                            val leafPathSize = leafPath?.path?.size
+                            var i = outlinePathSize.plus(1)
+                            if (leafPathSize != null) {
+                                while (i < leafPathSize - 1) {
+                                    val nextComp = leafPath.get(i)
+                                    environment.router.showCourseContainerOutline(
+                                            this@CourseDatesPageFragment, REQUEST_SHOW_COURSE_UNIT_DETAIL, courseData, null,
+                                            nextComp?.id, leafCompId, false)
+                                    i += 2
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
Can be moved in `OfflineSupportBaseFragment` call to make remove the duplicate code.
Also convert the `OfflineSupportBaseFragment` into kotlin class.